### ISSUE TITLE
FileNotFoundException running Python BasicTemplateAlgorithm

### ISSUE THREAD

**Issue by tomhunter-gh** [QuantConnect/Lean#915]
I've downloaded Lean as at https://github.com/QuantConnect/Lean/commit/0d118c89ba65a42a481a39a5bf6ed3a180430234 (2017-05-13), and I've run the below commands to switch from the UCS4 PythonNet assembly to the UCS2 one (as I'm on Windows 10):

```
sed -i -e 's/UCS4/UCS2/' Common/QuantConnect.csproj
sed -i -e 's/UCS4/UCS2/' Algorithm/QuantConnect.Algorithm.csproj
sed -i -e 's/UCS4/UCS2/' Algorithm.CSharp/QuantConnect.Algorithm.CSharp.csproj
sed -i -e 's/UCS4/UCS2/' Algorithm.Python/QuantConnect.Algorithm.Python.csproj
sed -i -e 's/UCS4/UCS2/' AlgorithmFactory/QuantConnect.AlgorithmFactory.csproj
```

I've updated the **config.json** file in the Launcher project to use Python:
```js
  // Algorithm language selector - options CSharp, FSharp, VisualBasic, Python, Java
  "algorithm-language": "Python",

  //Physical DLL location
  //"algorithm-location": "QuantConnect.Algorithm.CSharp.dll",
  "algorithm-location": "QuantConnect.Algorithm.Python.dll",
```

However when I run the Launcher application I get the following exception:
```
System.IO.FileNotFoundException was unhandled
  FileName=D:\Users\Tom\Dropbox\Algo Trading\QuantConnect\tomhunter-gh\Lean\Launcher\bin\Debug\QuantConnect.Algorithm.Python.dll
  HResult=-2147024894
  Message=Could not find file 'D:\Users\Tom\Dropbox\Algo Trading\QuantConnect\tomhunter-gh\Lean\Launcher\bin\Debug\QuantConnect.Algorithm.Python.dll'.
  Source=mscorlib
  StackTrace:
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.File.InternalReadAllBytes(String path, Boolean checkHost)
       at QuantConnect.Queues.JobQueue.NextJob(String& location) in D:\Users\Tom\Dropbox\Algo Trading\QuantConnect\tomhunter-gh\Lean\Queues\JobQueue.cs:line 128
       at QuantConnect.Lean.Launcher.Program.Main(String[] args) in D:\Users\Tom\Dropbox\Algo Trading\QuantConnect\tomhunter-gh\Lean\Launcher\Program.cs:line 77
  InnerException: 
```

From the build output it seems that the Python project is not configured to build with the default solution configuration:

> 15>------ Skipped Rebuild All: Project: QuantConnect.Algorithm.Python, Configuration: Debug Any CPU ------
> 15>Project not selected to build for this solution configuration

**Comment by tomhunter-gh** [comment]
Also wanted to mention that when I build the `QuantConnect.Algorithm.Python` project explicitly in Visual Studio, I get the following included in the output:

> 6>------ Build started: Project: QuantConnect.Algorithm.Python, Configuration: Debug Any CPU ------
> 6>  QuantConnect.Algorithm.Python -> D:\Users\Tom\Dropbox\Algo Trading\QuantConnect\tomhunter-gh\Lean\Algorithm.Python\bin\Debug\QuantConnect.Algorithm.Python.dll
> 6>  '´╗┐REM' is not recognized as an internal or external command,
> 6>  operable program or batch file.

However it looks like the project build (and the build.cmd script) complete successfully.

**Comment by tomhunter-gh** [comment]
It looks like the build.cmd script previously copied the `QuantConnect.Algorithm.Python.dll` assembly over to the Launcher bin directory: https://github.com/QuantConnect/Lean/commit/c1de2f768f02fc486da3605c8bfc89edd974f121

**Comment by tomhunter-gh** [comment]
Looks like the **QuantConnect.Algorithm.Python.dll** assembly is not needed.

I've changed my **config.json** as follows rather than what I had earlier:

```js
  //Physical DLL location
  //"algorithm-location": "QuantConnect.Algorithm.CSharp.dll",
  //"algorithm-location": "QuantConnect.Algorithm.Python.dll",
  "algorithm-location": "BasicTemplateAlgorithm.py",
```

I can now run the back test successfully.

**Comment by AlexCatarino** [comment]
Hi @tomhunter-gh, thank you for opening this issue.
We will remove references to **QuantConnect.Algorithm.Python.dll** since we cannot compile a dll with pythonnet, and modify the **config.json** with the path to the script. We can also point it to script in the **Algorithm.Python** project:
```
"algorithm-location": "../../../Algorithm.Python/BasicTemplateAlgorithm.py",
```
When pythonnet imports a module (pyc file), if it not found, it compiles the script (py file). At the moment, the batch file is only used to compile all the scripts to speed up the execution and check for errors

**Comment by jaredbroad** [comment]
Fixed in #916

**Comment by tomhunter-gh** [comment]
Excellent, thanks @AlexCatarino.  This enables easy debugging:

![image](https://cloud.githubusercontent.com/assets/1420946/26077458/0561aa68-39b4-11e7-8ea3-5b8285a9a593.png)

You have to first run without debugging and then, when you get the "Waiting for debugger..", attach to the `QuantConnect.Lean.Launcher.exe` process, selecting only Python code type:

![image](https://cloud.githubusercontent.com/assets/1420946/26077513/2df98a5e-39b4-11e7-94ea-d7fd664fb52f.png)
